add Autostart windows and messageBox timeout

Sav87 opened the pull request.

Sav87 committed:
Add TimeOut MsgBox

Sav87 committed:
fix AutoStart windows

Sav87 committed:
fix Autotart text and msgBox position

MSukharev commented:
Yes, NHML 1.8.0.0 is working. What difference?

MSukharev commented:
Someone wrote that if you wait 10 minutes after starting with an error and press start again, then the nicehash miner works. This is indeed so.

Crash911119 commented:
Очевидно, что проблема в файрволе со стороны сервера, клаудфайр блочит клиента на 10 минут. Надо понять, по каким признакам, клаудфайр идентифицирует клиента и поменять их, чтобы при следующем старте избежать бана. Все эти добавления таймеров в форму всего лишь костыль и дикий тормоз майнингна. Какие параметры найс передает клаудфайр?

MSukharev commented:
Интересно, что старые версии NHML работают без задержек, например, 1.8.0.0 (проверял). Так что проблемы не только со стороны сервера; это проблемы взаимодействия. Это может быть подсказкой для разработчика: что менялось в обмене со стратум-сервером между 1.8.0.0. и 1.8.1.5.

superyarik commented:
Но у меня нет 10 минут бана, просто несколько раз запускаю и на 5й 6й разы запускается, бан тут непричем

Sav87 commented:
Поэтому и нужно повторять подключение автоматически без блокировки сообщением. Скорее всего это не бан, просто клиент посылает кучу запросов на подключение до 20 в минуту с интервалом в секунду. Это создаёт нагрузку на сервер. Возможно поэтому сервер приостанавливает соодинение, и так по кругу. А клиентов много. Теперь насчёт что поменялось: раньше использовалось api запросы вида https://api.nicehash.com/api?.... а теперь используется вебсокет. Вполне нормальное решение только алгоритм не совсем удачный.

Sav87 commented:
Вечером постараюсь добавить резервный способ получения данных прибыльности

Olegik-ah commented:
По старинке через https://api.nicehash.com/api?method=simplemultialgo.info ?

Crash911119 commented:
По наблюдениям, бан со стороны сервера получаем только в случае некорректного завершения найса, тоесть когда машина упала и никто даже вякнуть не успел, с версией 1,7 такой проблемы вроде как нет. Если закрыть найс и перезапустить то все норм, может быть он при завершении просто посылает какую то команду закрытия сессии и если ее при следующей авторизации нет, то шлет клиента лесом??
ЗЫ я не проггер, исходники не смотрел

Sav87 commented:
Olegik-ah да.  Crash911119 Я больше склоняюсь к возросшей нагрузкой на сервер.

Crash911119 commented:
Sav87, если бы это была возросшая нагрузка на сервер, то просто перезапуск программы вызывал бы такую же ошибку, как и перезагрузка фермы, но если программу просто перезапустить, то все нормально

Sav87 commented:
Перезапускать тоже не всегда надо достаточно просто нажать старт поэтому я и добавил запуск по кругу

Sav87 committed:
receiving SMA via the API

Sav87 committed:
Merge branch 'NHconnection' into AutoStart

Olegik-ah commented:
Все работает супер! Берет по https, как оживает сокет - начинает брать через него. Несколько раз на разных ригах запускал - все стабильно!

Sav87 committed:
fix null

gromozeka666 commented:
Ребята, подскажите чего и где надо подправить? В програмировании ноль полный. Или у кого есть нормальная сборка? Скиньте на почту.

MSukharev commented:
Sav87, как я понимаю, нужно взять исходный текст майнера, исправить указанные Вами строки и компилировать? К сожалению, не все тут программисты. Я занимался когда-то давно, на Бэйсике и Паскале; наверно, придется вспомнить.

xtrime-ru commented:
@Sav87 Попросил товарища скомпилировать nicehashminerlegacy.exe с вашими правками - все работает отлично, огромное СПАСИБО!

Могу ли выложить скомпилированную программу с вашими правками, что бы остальные могли её использовать? Или nicehash может заблокировать старый api из за возросшей нагрузки?

Sav87 commented:
Как я могу запретить. Понятия не имею. Это api выложено на ихнем сайте.

xtrime-ru commented:
Я понимаю, что вы отношения к апи не имеете. Но никто скомпилированную программу не выкладывал, поэтому я и уточнил, есть ли какие то обьективные причины не выкладывать исправленную версию. Ну раз причин нет, то сейчас разберусь с гитхабом и выложу архив в течении нескольких часов. 

А то все форумы забиты танцами с бубном, как решить эту проблему :)

Sav87 commented:
Причин не выкладывать нет. Просто я предложил решение.

Crash911119 commented:
Выкладывай уже! а то пока Sav87 сиськи мнет со своей скромностью(хотя он и гениален!!!), мы с ума сойдем

xtrime-ru commented:
https://github.com/xtrime-ru/NiceHashMinerLegacy-SMA-FIX/releases/tag/1.8.1.5-sma-fix

exe без цифровой подписи, так что 10-ка ругается. Надо нажать "Подробнее" и потом "Все равно запустить". Чуть позже скриншоты и инструкцию сделаю.

Там не все файлы, а только необходимый минимум. Нужно скопировать в папку с 1.8.1.5 или 1.8.2-pre

xtrime-ru commented:
Сделал репозиторий: https://github.com/xtrime-ru/NiceHashMinerLegacy-SMA-FIX

gromozeka666 commented:
Огромное спасибо!!! РАБОТАЕТ!!! =)))
Фермочка маленькая, Win7x64, 3х1050TI, но снова в строю!
Сначала матюкнулось без высвечивания сообщения, потом запустилась и спустя несколько минут подтянула инфу. Супер! Ребята Вы ГЕНИИ!!!

Crash911119 commented:
Спасибо! работает!!! Ребята, вы сделали то, что разрабы уже несколько месяцев не могут.
С недавно вышедшей версией будет работать?

gromozeka666 commented:
на 1,8,1,5 работает, а вот 1,8,2 (оригинальная) у меня отказалась даже бенчмарк проходить, пропуски делает и всё, хоть об стену убейся...

xtrime-ru commented:
@gromozeka666 у меня официальная 1.8.2 тоже некоторые алгоритмы отказывается тестировать на некоторых картах, например 1080ti не хотят считать daggerhashimoto и все что с ним связано, в принципе не проблема там прибыли всеравно нет.

Попробуйте увеличить точность бенчмарка до стандартной или высокой точности.

Данные sma для бенчмарков вроде не требуются.

xtrime-ru commented:
@Crash911119 этот фикс по сути отдельная 1.8.1.5, даже если в папке стояла 1.8.2, то после замены будет 1.8.1.5 так что работать будет, но фишек из 1.8.2 не будет. 

Можно попробовать вручную внести исправления в ветку 1.8.2, но там сейчас много непроверенных коммитов, можно напороться на другие баги, легче дождаться релиза.

Sav87 commented:
Проблема с бенчмарком на нвидиа картах скорее всего связана проблемой завершения Майнера

Sav87 commented:
Исправления на ветку 1.8.2 должны встать.

Sav87 committed:
ConnectAsync and AutoStart

Asynchronous connection, to unlock the window.
Added time display before the first attempt and the total number of attempts

Sav87 committed:
25s. Add Icon to MsgBox

Sav87 committed:
AutoStart fix

Crash911119 commented:
1,8,2,0 сама по себе глючная. В чистом виде не всегда проходит бенчмарк на зеленых. Это не особо хорошо, так  как там уже встроен клеймор с криптонайтом (((. Всетаки ради стабильности придется остаться на 1,8,5,1 с фиксом на SMA/ спасибо  Sav87 за проделанную работу и xtrime-ru за компиляцию

angelbbs commented:
Покажите сырцы. По лицензии это обязательно!

xtrime-ru commented:
@angelbbs Исходники для компилированного exe это ветка master + смердженные коммиты до 1 декабря из этого пул реквеста. 

Завтра выложу исходники, если нужны. 
Или может @Sav87 скомпилирует из своей ветки, с новыми правками.

angelbbs commented:
Где решение проблемы получения SMA? В сырцах этого нет.

xtrime-ru commented:
@angelbbs я же уже присылал ссылку в ВК...

https://github.com/nicehash/NiceHashMinerLegacy/pull/625/commits/af3ef89cc788092717f0950e8787524890357ad0#diff-f60cd10b93fcf3b703cf7771847fe925R350 

350  строка, идет получение JSON строки по апи.

NHM_API_info прописан в файле /NiceHashMiner/Utils/Links.cs
https://github.com/Sav87/NiceHashMinerLegacy/blob/AutoStart/NiceHashMiner/Utils/Links.cs#L21

angelbbs commented:
спасибо. если есть желание, присоединяйтесь к ветке https://github.com/angelbbs/NiceHashMinerLegacy

Sav87 commented:
@angelbbs спасибо не надо. Чтобы на твой кошелёк работать? @xtrime-ru хотябы кошелёк не меняет, просто добавил на станице для пожертвований.

xtrime-ru commented:
@Sav87 Кстати кинь мне свой кошелек, скину тебе благодарность. Реально выручил меня :)

Sav87 commented:
33tudLZvY5SMuF2z1aUcMz8LKL55ZurDWk

xtrime-ru commented:
BTC?

Sav87 committed:
fix Async

Sav87 committed:
fix
